pkg/config/config_linux.go

```diff
@@ -24,3 +33,115 @@ func (c *RuntimeConfig) ValidatePinnsPath(executable string) error {
 
 	return err
 }
+
+// checkKernelRROMountSupport checks the kernel support for the Recursive Read-only (RRO) mounts.
+func checkKernelRROMountSupport() error {
+	kernelRROSupportOnce.Do(func() {
+		kernelRROSupportError = (func() error {
+			// Check the current kernel version for RRO mounts support...
+			err := validateKernelRROVersion()
+			if err != nil {
+				versionErr := err
+				// ... and if the kernel version does not match the minimum required version,
+				// then verify whether the kernel supports RRO mounts regardless, as often
+				// Linux distributions provide heavily patched kernel releases, and the
+				// current kernel might include backported support.
+				err = validateKernelRROMount()
+				if err != nil {
+					return fmt.Errorf("%v: %w", versionErr, err)
```

Comment:
nit: we can use `errors.Join` here.

Reply:
The problem with `errors.Join()` is this:

```
// The error formats as the concatenation of the strings obtained
// by calling the Error method of each element of errs, with a newline
// between each string.
```

This sadly breaks logrus per:

```console
WARN[2024-04-11 08:03:38.207720457Z] Runtime handler "runc" supports Recursive Read-only (RRO) mounts, but kernel does not: kernel version "5.10.214" does not support recursive read-only mounts
unable to set recursive read-only mount attribute: function not implemented
WARN[2024-04-11 08:03:38.215434604Z] Runtime handler "crun" supports Recursive Read-only (RRO) mounts, but kernel does not: kernel version "5.10.214" does not support recursive read-only mounts
unable to set recursive read-only mount attribute: function not implemented
```

This is an unfortunate API that Go has that does not allow you to customise the `Error()` method of the type that handles joining errors.

I can replace `%v` with `%w`, as from 1.20 onwards, `fmt.Errorf()` supports multiple error wrapping. This will have the same effect, sans the better error unwrapping if needed.